src/compiler/checker.ts

```diff
@@ -16122,7 +16129,7 @@ namespace ts {
 
             if (produceDiagnostics) {
                 checkTypeForDuplicateIndexSignatures(node);
-                checkUnusedTypeParameters(node);

```

Comment:
signatures do. they do not merge, so after looking at the signature, there is no need to wait, no more uses will show up after that.